flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/maintenance/operator/TestMonitorSource.java

```diff
@@ -0,0 +1,344 @@
+    long deleteSize =
+        withDelete
+            ? table
+                .currentSnapshot()
+                .addedDeleteFiles(table.io())
+                .iterator()
+                .next()
+                .fileSizeInBytes()
+            : 0;
+    assertThat(iterator.next())
+        .isEqualTo(new TableChange(1, withDelete ? 1 : 0, dataSize, deleteSize, 1));
+    // Make sure that consecutive calls do not return the data again
+    assertThat(iterator.next()).isEqualTo(EMPTY_EVENT);
+
+    // Add two more commits, but fetch the data in one loop
+    sql.exec("INSERT INTO %s VALUES (2, 'b')", TABLE_NAME);
+    table.refresh();
+    dataSize =
+        table.currentSnapshot().addedDataFiles(table.io()).iterator().next().fileSizeInBytes();
+    deleteSize =
+        withDelete
+            ? table
+                .currentSnapshot()
+                .addedDeleteFiles(table.io())
+                .iterator()
+                .next()
+                .fileSizeInBytes()
+            : 0;
+    sql.exec("INSERT INTO %s VALUES (3, 'c')", TABLE_NAME);
+    table.refresh();
+    dataSize +=
+        table.currentSnapshot().addedDataFiles(table.io()).iterator().next().fileSizeInBytes();
+    deleteSize +=
+        withDelete
+            ? table
+                .currentSnapshot()
+                .addedDeleteFiles(table.io())
+                .iterator()
+                .next()
+                .fileSizeInBytes()
+            : 0;
+    assertThat(iterator.next())
+        .isEqualTo(new TableChange(2, withDelete ? 2 : 0, dataSize, deleteSize, 2));
+    // Make sure that consecutive calls do not return the data again
+    assertThat(iterator.next()).isEqualTo(EMPTY_EVENT);
+  }
+
+  /**
+   * Create a table and check that the source returns the data as new commits arrive to the table.
+   */
+  @Test
+  void testSource() throws Exception {
+    sql.exec(
+        "CREATE TABLE %s (id int, data varchar) "
+            + "WITH ('flink.max-continuous-empty-commits'='100000')",
+        TABLE_NAME);
+    Configuration config = new Configuration();
+    config.set(CheckpointingOptions.CHECKPOINT_STORAGE, "filesystem");
+    config.set(CheckpointingOptions.CHECKPOINTS_DIRECTORY, "file://" + checkpointDir.getPath());
+    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+    env.enableCheckpointing(1000);
+    env.setParallelism(1);
+
+    TableLoader tableLoader = sql.tableLoader(TABLE_NAME);
+    tableLoader.open();
+    Table table = tableLoader.loadTable();
+    DataStream<TableChange> events =
+        env.fromSource(
+                new MonitorSource(tableLoader, HIGH_RATE, Long.MAX_VALUE),
+                WatermarkStrategy.noWatermarks(),
+                "TableChangeSource")
+            .forceNonParallel();
+
+    // Creating a stream for inserting data into the table concurrently
+    ManualSource<RowData> insertSource =
```

Comment:
wondering if we need the `ManualSource`. can we use the appender? e.g.

```
    GenericAppenderHelper dataAppender =
        new GenericAppenderHelper(tableResource.table(), FileFormat.PARQUET, TEMPORARY_FOLDER);
    List<Record> batch1 =
        RandomGenericData.generate(tableResource.table().schema(), 2, randomSeed.incrementAndGet());
    dataAppender.appendToTable(batch1);
```

Reply:
It will be used for testing the different MaintenanceTasks as well.
I wouldn't added the `ManualSource` for this use-case, but since it was added, I think reusing it here makes it easier to understand the real purpose of the test if we don't have to mix other helpers to the process.

Reply:
In this case, we are hooking up `ManualSource` with `FlinkSink` to insert records to the Iceberg table for `MonitorSource` if my understanding is correct. is `GenericAppenderHelper` simpler and faster for that purpose?

Reply:
Changed to `GenericAppenderHelper`, but kept the `ManualSource` as we will need it in the next PR again.